### PR TITLE
[posix] add netif SetState

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR adds the SetState for Netif module to set the interface ON/OFF.

In this PR this hasn't been integrated with NCP yet. (After receiving NCP property update, the state of wpan interface changed accordingly). This will be done later.

### CODE REVIEW

File: src/ncp/posix/netif.cpp [hunk 1]
```diff
@@ -114,6 +114,35 @@ void Netif::UpdateIp6UnicastAddresses(const std::vector<Ip6AddressInfo> &aAddrIn
...
+
+    VerifyOrExit(mIpFd >= 0);
+    memset(&ifr, 0, sizeof(ifr));
+    strncpy(ifr.ifr_name, mNetifName.c_str(), IFNAMSIZ - 1);
+    VerifyOrExit(ioctl(mIpFd, SIOCGIFFLAGS, &ifr) == 0, error = OTBR_ERROR_ERRNO);
+
+    ifState = ((ifr.ifr_flags & IFF_UP) == IFF_UP) ? true : false;
+
+    otbrLogNotice("[netif] Changing interface state to %s%s.", aState ? "up" : "down",
+                  (ifState == aState) ? " (already done, ignoring)" : "");
```
Comment (zhanglongxia) [inline src/ncp/posix/netif.cpp:131]: Why do we need to use NOTE log level? Could we just use INFO log level?

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/ncp/posix/netif.cpp [hunk 2]
```diff
@@ -114,6 +114,35 @@ void Netif::UpdateIp6UnicastAddresses(const std::vector<Ip6AddressInfo> &aAddrIn
...
+    bool         ifState = false;
+
+    VerifyOrExit(mIpFd >= 0);
+    memset(&ifr, 0, sizeof(ifr));
+    strncpy(ifr.ifr_name, mNetifName.c_str(), IFNAMSIZ - 1);
+    VerifyOrExit(ioctl(mIpFd, SIOCGIFFLAGS, &ifr) == 0, error = OTBR_ERROR_ERRNO);
+
+    ifState = ((ifr.ifr_flags & IFF_UP) == IFF_UP) ? true : false;
+
+    otbrLogInfo("[netif] Changing interface state to %s%s.", aState ? "up" : "down",
```
Comment (jwhui) [inline src/ncp/posix/netif.cpp:130]: Is the `[netif]` prefix common in this project? We should standardize on an approach.

Comment (Irving-cl) [reply src/ncp/posix/netif.cpp]: Removed `[netif]` since we already have the log tag `NETIF` in otbr. (The log prefix is copied from OT posix netif).

**Review by jwhui — APPROVED** [review]

**Pull request merged**